### ISSUE TITLE
default_ip_address and guest_ip_addresses are not updated/refreshed after guest IP changes

### ISSUE THREAD

**Issue by dannyk81** [terraform-providers/terraform-provider-vsphere#671]
### Terraform Version

`Terraform v0.11.8`

### vSphere Provider Version

`v1.9.0`

### Affected Resource(s)

- `vsphere_virtual_machine`

### Expected Behavior

When the IP address of the Guest has changed, we expect to see the new IP in both the `default_ip_address` and `guest_ip_addresses` attributes.

### Actual Behavior

After running a terraform `plan` we executed a terraform `show` and the state still shows the "old" IP that this VM received during initial provisioning.

### Steps to Reproduce

1. `terraform apply`
2. VM gets provisioned and gets an IP via DHCP
3. Change the IP (either via DHCP or statically)
4. `terraform apply` --> `terraform show`
5. Observe that the IP as reported by `default_ip_address` and/or `guest_ip_addresses` is not updated

### Important Factoids

- We use DHCP to provision IPs for VMs, however in various cases it is possible that the IP will change, for example: if we move the VM to a different Network.
- We use the `default_ip_address` for an outpout variable, that is used as an input for a DNS module that sets up the relevant DNS records via the powerdns provider, since the `default_ip_address` doesn't get updated, the related dns records don't get updated as well.

**Comment by bill-rich** [comment]
Hi @dannyk81! Can you confirm that guest tools are installed and working on the VMs and that the new IP addresses are showing up in VMware before running `terraform apply`? If so, would it be possible to get the debug logs from a run where the IP address is not being updated? I'd like to see if there is any info on why that step is being skipped on `Read`.

**Comment by dannyk81** [comment]
Hi @bill-rich! indeed open-vm-tools is running and the correct IP address shows up in vSphere Client.

![image](https://user-images.githubusercontent.com/24274156/49776257-46b9ae00-fcc9-11e8-9a6d-a492f6459b24.png)


Here's a snippet of the `terraform show` command (after running plan/apply):
```
module.lbchat.vsphere_virtual_machine.srv.1:
  id = 4214ef3a-43ac-8701-7e37-ccb595e7fb17
  default_ip_address = 10.xxx.xxx.110
  guest_ip_addresses.# = 1
  guest_ip_addresses.0 = 10.xxx.xxx.110
```

The IP should be `10.xxx.xxx.113`, but it still shows `10.xxx.xxx.110`

I'm attaching a debug log as requested (I skipped all the graph walks, hope this is enough).

[debug.txt](https://github.com/terraform-providers/terraform-provider-vsphere/files/2665805/debug.txt)

I took a quick look at the code and unless I missed something I couldn't find where these attributes are actually refreshed/updated after initial create.
